docs/content/guides/columns/column-filter.md

```diff
@@ -155,765 +450,2860 @@ import 'handsontable/dist/handsontable.full.min.css';
+            model: 'Racing Socks',
+            price: 30,
+            sellDate: 'Oct 11, 2023',
+            sellTime: '01:23 AM',
+            inStock: false,
+          },
+          {
+            brand: 'Gigabox',
+            model: 'HL Mountain Frame',
+            price: 1890.9,
+            sellDate: 'May 3, 2023',
+            sellTime: '11:27 AM',
+            inStock: false,
+          },
+          {
+            brand: 'Camido',
+            model: 'Cycling Cap',
+            price: 130.1,
+            sellDate: 'Mar 27, 2023',
+            sellTime: '03:17 AM',
+            inStock: true,
+          },
+          {
+            brand: 'Chatterpoint',
+            model: 'Road Tire Tube',
+            price: 59,
+            sellDate: 'Aug 28, 2023',
+            sellTime: '08:01 AM',
+            inStock: true,
+          },
+          {
+            brand: 'Eidel',
+            model: 'HL Road Tire',
+            price: 279.99,
+            sellDate: 'Oct 2, 2023',
+            sellTime: '01:23 AM',
+            inStock: true,
+          },
+        ]}
+        columns={[
+          {
+            title: 'Brand',
+            type: 'text',
+            data: 'brand',
+          },
+          {
+            title: 'Model',
+            type: 'text',
+            data: 'model',
+          },
+          {
+            title: 'Price',
+            type: 'numeric',
+            data: 'price',
+            numericFormat: {
+              pattern: '$ 0,0.00',
+              culture: 'en-US',
+            },
+            className: 'htLeft',
+          },
+          {
+            title: 'Date',
+            type: 'date',
+            data: 'sellDate',
+            dateFormat: 'MMM D, YYYY',
+            correctFormat: true,
+            className: 'htRight',
+          },
+          {
+            title: 'Time',
+            type: 'time',
+            data: 'sellTime',
+            timeFormat: 'hh:mm A',
+            correctFormat: true,
+            className: 'htRight',
+          },
+          {
+            title: 'In stock',
+            type: 'checkbox',
+            data: 'inStock',
+            className: 'htCenter',
+          },
+        ]}
+        filters={true}
+        height="auto"
+        licenseKey="non-commercial-and-evaluation"
+      />
+    </>
+  );
+};
 
-    return dataAtCol.filter((value, index, self) => self.indexOf(value) === index);
-  }
+ReactDOM.render(<App />, document.getElementById('exampleFilterAsYouType'));
+```
 
-  getSourceDataAtCol(column) {
-    return this.data.map(dataAtRow => dataAtRow[column].toString());
-  }
+:::
 
-  setSelectedColumn(column) {
-    this.selectedColumn = column;
-    this.initStateForColumn();
-  }
+:::
 
-  getSelectedColumn() {
-    return this.selectedColumn;
-  }
+## Customize the filter button
 
-  getValuesToFilter() {
-    return this.valuesToFilter;
-  }
+The filter menu is part of the [column menu](@/guides/columns/column-menu.md), so they both use the
```

Comment:
The filter menu is not part of the column menu (same case as previously), so I would just skip that part.

This section should be similar to https://handsontable.com/docs/react-data-grid/rows-sorting/#add-custom-sort-icons, maybe:

> The default button that opens the column menu can be styled with CSS by modifying `button.changeType` and its `::before` pseudoclass that contains an HTML entity displaying an arrow down icon.

I would change the style of the buttons in the demo slightly:

```css
.custom-filter-button-example-1 .changeType {
    background: #e2e2e2;
    border-radius: 100%;
    width: 16px;
    color: #0075ff;
    height: 16px;
    padding: 3px 2px;
    border: none;
}

.custom-filter-button-example-1 .changeType::before {
    content: "▼ ";
    zoom: .9;
}
```

The result:
<img width="790" alt="Screenshot 2023-04-12 at 22 48 17" src="https://user-images.githubusercontent.com/8048526/231581784-ed3a3240-e2ff-4b72-bf69-074bbb68ee32.png">